src/OrbitBase/Logging.cpp

```diff
@@ -7,56 +7,107 @@
 #include <array>
 #include <system_error>
 
+#include "Path.h"
 #include "absl/base/const_init.h"
 #include "absl/debugging/stacktrace.h"
 #include "absl/debugging/symbolize.h"
+#include "absl/flags/flag.h"
 #include "absl/strings/str_cat.h"
 #include "absl/strings/str_format.h"
 #include "absl/synchronization/mutex.h"
+#include "absl/time/time.h"
 
+ABSL_FLAG(std::string, log_dir, "", "Set directory for the log.");
 static absl::Mutex log_file_mutex(absl::kConstInit);
 std::ofstream log_file;
-const int kLogFileLifetimeMinutes = 60 * 24 * 7;  // one week
+const absl::Duration kLogFileLifetime = absl::Hours(24 * 7);  // one week
 
-std::optional<int> GetLogFileLifetimeMinutes(std::string log_file_name) {
+namespace orbit_base {
+[[nodiscard]] uint32_t GetCurrentProcessId() {
+#ifdef _WIN32
+  uint32_t current_pid = static_cast<uint32_t>(::GetCurrentProcessId());
+#else
+  uint32_t current_pid = static_cast<uint32_t>(getpid());
+#endif
+  return current_pid;
+}
+
+[[nodiscard]] std::filesystem::path CreateOrGetLogDir() {
+  std::filesystem::path logs_dir;
+  if (!absl::GetFlag(FLAGS_log_dir).empty()) {
+    logs_dir = absl::GetFlag(FLAGS_log_dir);
+  } else {
+    logs_dir = Path::CreateOrGetOrbitAppDataDir() / "logs";
+  }
+  std::filesystem::create_directory(logs_dir);
+  return logs_dir;
+}
+
+std::filesystem::path GetLogFilePath(const absl::Time& timestamp, const uint32_t& pid) {
+  std::filesystem::path logs_dir = CreateOrGetLogDir();
+  std::string timestamp_string =
+      absl::FormatTime(kLogFileNameTimeFormat, timestamp, absl::LocalTimeZone());
+  std::string filename = absl::StrCat("Orbit", kLogFileNameDelimiter, timestamp_string,
+                                      kLogFileNameDelimiter, pid, ".log");
+  return logs_dir / filename;
+}
+}  // namespace orbit_base
+
+std::vector<std::filesystem::path> ListFiles(const std::filesystem::path& log_dir) {
+  std::vector<std::filesystem::path> files_in_dir;
+  for (const auto& file_path : std::filesystem::recursive_directory_iterator(log_dir)) {
+    if (file_path.is_regular_file()) {
+      files_in_dir.push_back(file_path.path());
+    }
+  }
+  return files_in_dir;
+}
+
+ErrorMessageOr<absl::Duration> GetLogFileLifetime(const std::string& log_file_name) {
   unsigned first_delimiter_pos = log_file_name.find_first_of(kLogFileNameDelimiter);
```

Comment:
From https://www.cplusplus.com/reference/string/string/find_first_of/
> The position of the first character that matches.
If no matches are found, the function returns string::npos.

Have an early return with ErrorMessage here, when `kLogFileNameDelimiter` is not found. Or in other cases where there is something wrong. 

The longer I think about this. Maybe you can make this functions even stronger. The log file name should always have this format: `Orbit-%time%-%pid%.log`. That means the time always starts at character 6. Also the length of the time is fixed.. right? so no need to calculate the length of the time

Reply:
Make sense! I changed to use constants `kTimestampStartPos` and `kTimestampStringLength` to indicate the starting position and the timestamp string length. Thanks!

Reply:
Looks great :)